internal/storage/storage.go

```diff
@@ -0,0 +1,117 @@
+package storage
+
+import (
+	"errors"
+	"sync"
+)
+
+type URLData struct {
+	URL string
+	ID  string
+}
+
+type SafeMap struct {
+	mu             sync.Mutex
+	mapStor        map[string]URLData
+	reverseMapStor map[URLData]string
+}
+
+var _ Repositories = (*TempStorage)(nil)
+var sm = SafeMap{}
```

Comment:
CHANGE REQUEST: please don't use global variables

Reply:
1. `sm` is a variable.
2. You can create a struct and return an interface.

Reply:
done